File: Leetle/tests/reference_solutions/python/reverse_string.py
def reverse_string(s):
    """
    Write a function that reverses a string. The input string is given as an array of characters s.
    You must do this by modifying the input array in-place.
    """
    left, right = 0, len(s) - 1
    while left < right:
        s[left], s[right] = s[right], s[left]
        left += 1
        right -= 1

# Main function to handle input from stdin
if __name__ == "__main__":
    import sys
    import ast
    input_data = sys.stdin.read().strip()
    s = ast.literal_eval(input_data)
    reverse_string(s)
    print(str(s).replace(" ", ""))

File: Leetle/tests/reference_solutions/javascript/reverse_string.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
function reverseString(s) {
    let left = 0, right = s.length - 1;
    while (left < right) {
        [s[left], s[right]] = [s[right], s[left]];
        left++;
        right--;
    }
}

// Main function to handle input from stdin
const fs = require('fs');
const input = fs.readFileSync(0, 'utf-8').trim();
const s = JSON.parse(input);

reverseString(s);
console.log(JSON.stringify(s));
